File: SimpleHabit/app/src/main/java/com/me/simplehabit/persistence/SimpleHabitDatabase.java
package com.me.simplehabit.persistence;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.arch.persistence.room.TypeConverters;
import android.content.Context;

import com.me.simplehabit.data.vos.CategoriesProgramVO;
import com.me.simplehabit.data.vos.CurrentProgramVO;
import com.me.simplehabit.data.vos.TopicVO;
import com.me.simplehabit.persistence.dao.CategoryProgramDao;
import com.me.simplehabit.persistence.dao.CurrentProgramDao;
import com.me.simplehabit.persistence.dao.TopicDao;
import com.me.simplehabit.persistence.typeconverters.IntegerListConverter;
import com.me.simplehabit.persistence.typeconverters.ProgramListConverter;
import com.me.simplehabit.persistence.typeconverters.SessionListConverter;

import java.util.List;

@Database(entities = {CurrentProgramVO.class,TopicVO.class, CategoriesProgramVO.class},version = 3)
@TypeConverters({SessionListConverter.class, IntegerListConverter.class, ProgramListConverter.class})
public abstract class SimpleHabitDatabase extends RoomDatabase {

    private static SimpleHabitDatabase objInstance;
    private static final String DB_NAME="simple_habit.db";

    public abstract CurrentProgramDao getCurrentProgramDao();
    public abstract TopicDao getTopicDao();
    public abstract CategoryProgramDao getCategoryProgramDao();

    public static SimpleHabitDatabase getDatabase(Context context){

        if (objInstance==null){
            objInstance= Room.databaseBuilder(context,SimpleHabitDatabase.class,DB_NAME)
                    .fallbackToDestructiveMigration()
                    .allowMainThreadQueries()
                    .build();
        }

        return objInstance;
    }

    public boolean isCurrentProgramEmpty() {
        CurrentProgramVO currentProgram = getCurrentProgramDao().retrieveCurrentProgram();
        return currentProgram == null;
    }

    public boolean isTopicEmpty(){
        List<TopicVO> topicList=getTopicDao().retrieveTopics();
        return topicList.isEmpty();
    }

    public boolean isCategoiresAndProgramsEmpty(){
        List<CategoriesProgramVO> list=getCategoryProgramDao().retrieveCategoriesAndPrograms();
        return list.isEmpty();
    }
}

File: SimpleHabit/app/src/main/java/com/me/simplehabit/persistence/SimpleHabitDatabase.kt
package com.me.simplehabit.persistence

import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import android.content.Context

import com.me.simplehabit.data.vos.CategoriesProgramVO
import com.me.simplehabit.data.vos.CurrentProgramVO
import com.me.simplehabit.data.vos.TopicVO
import com.me.simplehabit.persistence.dao.CategoryProgramDao
import com.me.simplehabit.persistence.dao.CurrentProgramDao
import com.me.simplehabit.persistence.dao.TopicDao
import com.me.simplehabit.persistence.typeconverters.IntegerListConverter
import com.me.simplehabit.persistence.typeconverters.ProgramListConverter
import com.me.simplehabit.persistence.typeconverters.SessionListConverter

@Database(entities = [
    CurrentProgramVO::class,
    TopicVO::class,
    CategoriesProgramVO::class], version = 3)
@TypeConverters(SessionListConverter::class, IntegerListConverter::class, ProgramListConverter::class)
abstract class SimpleHabitDatabase : RoomDatabase() {

    abstract val currentProgramDao: CurrentProgramDao
    abstract val topicDao: TopicDao
    abstract val categoryProgramDao: CategoryProgramDao

    val isCurrentProgramEmpty: Boolean
        get() {
            val count = currentProgramDao.retrieveCurrentProgramCount()
            return count < 1
        }

    val isTopicEmpty: Boolean
        get() {
            val topicList = topicDao.retrieveTopicsCount()
            return topicList < 1
        }

    val isCategoriesAndProgramsEmpty: Boolean
        get() {
            val list = categoryProgramDao.retrieveCategoriesAndProgramsCount()
            return list < 1
        }

    companion object {

        private var objInstance: SimpleHabitDatabase? = null
        private const val DB_NAME = "simple_habit.db"

        fun getDatabase(context: Context): SimpleHabitDatabase {

            if (objInstance == null) {
                objInstance = Room.databaseBuilder(context, SimpleHabitDatabase::class.java, DB_NAME)
                        .fallbackToDestructiveMigration()
                        .allowMainThreadQueries()
                        .build()
            }

            return objInstance as SimpleHabitDatabase
        }
    }
}